webdriver/tests/take_screenshot/user_prompts.py

```diff
@@ -2,63 +2,104 @@
 
 import pytest
 
-from tests.support.asserts import assert_png, assert_success
+from tests.support.asserts import assert_png, assert_success, assert_error
 from tests.support.inline import inline
 
 
 def take_screenshot(session):
     return session.transport.send(
         "GET", "session/{session_id}/screenshot".format(**vars(session)))
 
+# In some cases, such as "dismiss and notify" & "accept and notify" dismiss
+# the dialog and throw an error. This fixture checks those cases.
+@pytest.fixture
+def check_user_prompt_closed_with_exception(session, create_dialog):
```

Comment:
You would probably be able to share this between tests if you put it in https://github.com/web-platform-tests/wpt/blob/master/webdriver/tests/support/fixtures.py.

Reply:
I made some attempts at doing this and ran into some complications:

- take_element_screenshot and take_screenshot differ in that the element variant requires an element ID to take a screenshot of, this complicates the new proposed derived replacement function.
- fixtures.py does not currently have assert functions and inline functions needed by check_user_prompt_closed_with_error, putting this there feels kind of out of place as the other functions seem to be things that help out with tests rather than a generic version of a test.

If you can think of a better way to do this let me know, otherwise I would propose leaving this as is - I think they are different enough to justify a similar copy in two places.

Reply:
Oh right, thanks for pointing that out. Having one separate for each file makes sense to avoid that complexity.